### PR TITLE
[SPARK-30802][ML] Use Summarizer instead of MultivariateOnlineSummarizer in Aggregator test suite

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
There are three changes in this PR:
1. use Summarizer instead of MultivariateOnlineSummarizer in Aggregator test suites (similar to https://github.com/apache/spark/pull/26396)
2. Put common code in ```Summarizer.getRegressionSummarizers``` and ```Summarizer.getClassificationSummarizers```.
3. Move ```MultiClassSummarizer``` from ```LogisticRegression``` to ```ml.stat``` (this seems to be a better place since ```MultiClassSummarizer``` is not only used by ```LogisticRegression``` but also several other classes).

### Why are the changes needed?
Minimize code duplication and improve performance

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
existing test suites.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala [hunk 1]
```diff
@@ -202,6 +204,38 @@ object Summarizer extends Logging {
     val (metrics, computeMetrics) = getRelevantMetrics(requested)
     new SummarizerBuffer(metrics, computeMetrics)
   }
+
+  /** Get regression feature and label summarizers for provided data. */
+  private[spark] def getRegressionSummarizers(
```
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala]: `private[ml]` ?

Comment (huaxingao) [reply mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala]: Thanks! Updated.

File: mllib/src/main/scala/org/apache/spark/ml/stat/MultiClassSummarizer.scala [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+   * @param label The label for this data point.
+   * @param weight The weight of this instances.
+   * @return This MultilabelSummarizer
+   */
+  def add(label: Double, weight: Double = 1.0): MultiClassSummarizer = {
+    require(weight >= 0.0, s"instance weight, $weight has to be >= 0.0")
+
+    if (weight == 0.0) return this
+
+    if (label - label.toInt != 0.0 || label < 0) {
```
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/stat/MultiClassSummarizer.scala]:
```scala
    if (label - label.toInt != 0.0 || label < 0) {
      totalInvalidCnt += 1
    } else {
      val (counts: Long, weightSum: Double) = distinctMap.getOrElse(label.toInt, (0L, 0.0))
      distinctMap.put(label.toInt, (counts + 1L, weightSum + weight))
    }
    this


```

File: mllib/src/main/scala/org/apache/spark/ml/stat/MultiClassSummarizer.scala [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    largeMap.totalInvalidCnt += smallMap.totalInvalidCnt
+    largeMap
+  }
+
+  /** @return The total invalid input counts. */
+  def countInvalid: Long = totalInvalidCnt
+
+  /** @return The number of distinct labels in the input dataset. */
+  def numClasses: Int = if (distinctMap.isEmpty) 0 else distinctMap.keySet.max + 1
```
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/stat/MultiClassSummarizer.scala]: nit: `distinctMap.keysIterator.max + 1`

**Review by zhengruifeng — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by huaxingao** [comment]
Thanks! @srowen @zhengruifeng

**Pull request closed without merging**